tests/regression/test_eigensolver.py

```diff
@@ -0,0 +1,54 @@
+import numpy as np
+import pytest
+from firedrake import *
+
+
+def evals(n, degree=1, mesh=None):
+    '''We base this test on the 1D Poisson problem with Dirichlet boundary
+    conditions, outlined in part 1 of Daniele Boffi's
+    'Finite element approximation of eigenvalue problems' Acta Numerica 2010'''
+    # Create mesh and define function space
+    if mesh is None:
+        mesh = IntervalMesh(n, 0, pi)
+    V = FunctionSpace(mesh, "CG", degree)
+    # Define variational problem
+    u = TrialFunction(V)
+    v = TestFunction(V)
+    a = (inner(grad(u), grad(v))) * dx
+
+    # Create eigenproblem with boundary conditions
+    bc = DirichletBC(V, 0.0, "on_boundary")
+    eigenprob = LinearEigenproblem(a, bcs=bc, bc_shift=-6666.)
+
+    # Create corresponding eigensolver, looking for n eigenvalues
+    eigensolver = LinearEigensolver(
+        eigenprob, n, solver_parameters={"eps_largest_real": None}
+    )
+    ncov = eigensolver.solve()
+
+    # boffi solns
+    h = pi / n
+    true_values = np.zeros(ncov-1)
+    estimates = np.zeros(ncov-1)
+    for k in range(ncov-1):
+        true_val = 6 / h**2
+        # k+1 because we skip the trivial 0 eigenvalue
+        true_val *= (1-cos((k+1)*h))/(2+cos((k+1)*h))
+        true_values[k] = true_val
+
+        estimates[k] = eigensolver.eigenvalue(k).real
+
+    true_values[-1] = eigenprob.bc_shift
+
```

Comment:
why are you not also shifting the 0 entry?


and, is it safe to assume this numbering?

Reply:
The 0 entry is not a boundary entry. The other boundary is off the end of the array (i.e. we don't look at it).

I think the ordering is safe because of the solver option to look for largest values (at any rate, the observation is that it seems stable).